pkg/deploy/sync.go

```diff
@@ -28,13 +28,23 @@ import (
 
 // Sync syncs the blueprint to the cluster in a generic (as much as Go allows) manner.
 // Returns true if object is up to date otherwiser returns false
+//
+// WARNING: For legacy reasons, this method bails out quickly without doing anything if the CheCluster resource
+// is being deleted (it does this by examining the deployContext, not the cluster). If you don't want
+// this behavior, use the DoSync method.
 func Sync(deployContext *DeployContext, blueprint metav1.Object, diffOpts ...cmp.Option) (bool, error) {
 	// eclipse-che custom resource is being deleted, we shouldn't sync
 	// TODO move this check before `Sync` invocation
 	if !deployContext.CheCluster.ObjectMeta.DeletionTimestamp.IsZero() {
 		return true, nil
 	}
 
+	return DoSync(deployContext, blueprint, diffOpts...)
+}
+
+// Sync syncs the blueprint to the cluster in a generic (as much as Go allows) manner.
+// Returns true if object is up to date otherwiser returns false
+func DoSync(deployContext *DeployContext, blueprint metav1.Object, diffOpts ...cmp.Option) (bool, error) {
```

Comment:
Why do we need to sync if checluster is being deleted. What is a case?

Reply:
I just wanted to use this in a generic way and the shortcut logic in a generic sync method seemed a little bit out of place to me, following the "principle of least astonishment". Otherwise you're right, all the callers in the current codebase should behave equivalently regardless of the method used.